src/Infiltration/ui/SlashGame.tsx

```diff
@@ -25,48 +25,52 @@ const difficulties: {
   Impossible: { window: 150 },
 };
 
-export function SlashGame(props: IMinigameProps): React.ReactElement {
+export function SlashGame({ difficulty: _difficulty, onSuccess, onFailure }: IMinigameProps): React.ReactElement {
   const difficulty: Difficulty = { window: 0 };
-  interpolate(difficulties, props.difficulty, difficulty);
+  interpolate(difficulties, _difficulty, difficulty);
+
   const [phase, setPhase] = useState(0);
 
   function press(this: Document, event: KeyboardEvent): void {
     event.preventDefault();
     if (event.key !== KEY.SPACE) return;
     if (phase !== 1) {
-      props.onFailure();
+      onFailure();
     } else {
-      props.onSuccess();
+      onSuccess();
     }
   }
-  const hasAugment = Player.hasAugmentation(AugmentationName.MightOfAres, true);
-  const guardingTime = Math.random() * 3250 + 1500 - (250 + difficulty.window);
-  const preparingTime = difficulty.window;
-  const attackingTime = 250;
+
+  const guardingTimeRef = useRef(Math.random() * 3250 + 1500 - (250 + difficulty.window));
```

Comment:
this is one of the places where you can barely get rid of `useRef` - here variable is supposed to be generated only once, during the first render iteration. And then it is passed down both to child component and to `useEffect` (so we can't move it inside useEffect).
An alternative could be React state, but to me it sounds even worse as state is something you expect to change at some point. If you used React class-based component, you would declare a field in this class and initialize it in the constructor (cause you want this value to be set only once). That's effectively the same as using ref in function components.

The only problem I see here is that if the same game could be used twice in a row - this ref won't be recalculated leading to the same guard time value... But this PR does not change this part, runtime result should be the same as before.

Reply:
> Also in some cases you just have to use refs.

Yeah, there are a lot of new refs that I'm totally fine with. Honestly, due to the way our components work there's some that should be using less state and more refs, even though that's backwards for React.

What I was questioning was spots where a ref was introduced and never changed from its default value. That's just misuse IMO, created to appease the lint. (Likely the useEffect it's satisfying is also bad/weird.)

Reply:
> The only problem I see here is that if the same game could be used twice in a row - this ref won't be recalculated leading to the same guard time value... But this PR does not change this part, runtime result should be the same as before.

On that point, infiltration alternates between a game and a countdown timer, so the old game has already been totally unmounted by the time the next game is rendered. Even if that gets changed at some point, the game remembers the last game played and doesn't do the same game twice in a row. So each new game is always a new component being rendered, not just a component from last game getting updated.

> What I was questioning was spots where a ref was introduced and never changed from its default value.

The question is, what should be used in those cases? Previously in these cases we were using state, and just never using the setState function. Both state and ref in these cases doesn't really seem right. But what to use for something that is calculated within the component on first render, never changes, and can't be contained within a useEffect?

Reply:
Or just compute the value outside and capture it.
If it depends on props, then it's a bit sus (and useMemo probably calls that out the best)